packages/flutter_bloc/lib/src/bloc_builder.dart

```diff
@@ -145,6 +155,7 @@ class _BlocBuilderBaseState<B extends BlocBase<S>, S>
 
   @override
   Widget build(BuildContext context) {
+    if (widget.bloc == null) context.select<B, int>(identityHashCode);
```

Comment:
It's a very creative idea to solve the previous issue! I made one minor adjustment to use `identityHashCode` in case developers override `hashCode` manually.